src/ParseServer.js

```diff
@@ -87,9 +87,18 @@ class ParseServer {
       .performInitialization()
       .then(() => hooksController.load())
       .then(async () => {
+        const startupPromises = [];
         if (schema) {
-          await new DefinedSchemas(schema, this.config).execute();
+          startupPromises.push(new DefinedSchemas(schema, this.config).execute());
         }
+        if (
+          options.cacheAdapter &&
+          options.cacheAdapter.connect &&
+          typeof options.cacheAdapter.connect === 'function'
+        ) {
+          startupPromises.push(options.cacheAdapter.connect());
+        }
+        await Promise.all(startupPromises);
```

Comment:
I'm not sure about the `promise.all` since `DefinedSchemas` could involve "cacheAdapter".
Some things could mess up.

I can suggest first initializing the cache and then running defined schemas. 

I'm not 100% sure, but DefinedSchema use Parse schema under the hood and internal mechanism of parse, so there is an high chance that DefinedSchema feature use `cacheAdapter` internally.

I think it's safer and more reliable in the long term here to perform a sequential init with cache and then defined schema

Reply:
Thank you for the review!

DefinedSchemas affects the DB whereas cacheAdapter affects Redis, memory store etc, which is why I think it’s safe to run in parallel

Reply:
okay @dblythy i checked, it seems that Schemas use `SchemaCache` and this is not related to the common cache adapter.

@dblythy at this time the Defined Schema seems to do not use Schema Adapter BUT, we need to cover that in the futur the `cacheAdapter` is not used during DefinedSchema, otherwise weird issues will appear.

I can suggest adding a test on DefinedSchema, and you call directly "new DefinedSchema()", then you spy that 0 calls were detected on connect, get, and other public methods of the `cacheAdapter`. It will ensure that in the futur Defined Schema will stay stable 🚀

Reply:
also this test will cover my feedback since we are not 100% if the DefinedSchema feature is fully isolated from the CacheAdapter

Reply:
I added a test to cover the feature